src/systemcmds/topic_listener/listener_main.cpp

```diff
@@ -113,6 +113,9 @@ void listener(const orb_id_t &id, unsigned num_msgs, int topic_instance,
 		fds[1].fd = sub;
 		fds[1].events = POLLIN;
 
+		// Clear screen
+		dprintf(1, "\033[2J\n");
```

Comment:
Maybe only if `num_msgs > 1`? Or alternatively make sure we only fall into this case if it's going to be more than 1?

or catch it here perhaps? https://github.com/PX4/PX4-Autopilot/pull/23986/files#diff-95ea6d47bf95b7f58b56e2f57e462a07eee1d497fa4c1ff7f019f8e704470db6R59

Reply:
That's already the case.

We're in the `else` of `if (topic_instance == -1 && num_msgs == 1) {`

Reply:
If you specify only the instance it's going to hit this case.

Reply:
I think that's wrong, but I suppose we can take another pass later if anyone actually cares.

Reply:
Oh, I see.

Reply:
Done: https://github.com/PX4/PX4-Autopilot/pull/24019